packages/core/src/lib/GooglePlay.ts

```diff
@@ -62,12 +63,104 @@ export class GooglePlay {
    * https://github.com/Triple-T/gradle-play-publisher#uploading-a-pre-existing-artifact
    * @param track - Specifies the track that the user would like to publish to.
    */
-  async publishBundle(track: PlayStoreTrack, filepath: string): Promise<void> {
+  async publishBundleWithGradle(track: PlayStoreTrack, filepath: string): Promise<void> {
     // Uploads the artifact to the default internal track.
     await this.gradleWrapper.executeGradleCommand(
         ['publishBundle', '--artifact-dir', filepath, '--track', track]);
   }
 
+  /**
+   * This calls the publish bundle command and publishes an existing artifact to Google
+   * Play.
+   * Calls the following Play API commands in order:
+   * Edits.Insert
+   * Edits.Bunldes.Upload
+   * Edits.Tracks.Update
+   * Edits.Commit
+   * https://developers.google.com/android-publisher/api-ref/rest/v3/edits.bundles/upload
+   * https://developers.google.com/android-publisher/edits#workflow
+   * @param track - Specifies the track that the user would like to publish to.
+   * @param filepath - Filepath of the App bundle you would like to upload.
+   * @param packageName - packageName of the bundle.
+   * @param retainedBundles - all bundles that should be retained on upload. This is useful for
+   *   ChromeOS only releases.
+   */
+  async publishBundle(
+      track: PlayStoreTrack,
+      filepath: string,
+      packageName: string,
+      retainedBundles: number[],
+  ): Promise<void> {
+    // TODO(@nohe427): Remove this check when refactor is finished.
+    if (!this._googlePlayApi) {
+      return;
+    }
+    const edit = await this._googlePlayApi.edits.insert({packageName: packageName});
+    const editId = edit.data.id;
+    const result = await this._googlePlayApi.edits.bundles.upload(
+        {
+          ackBundleInstallationWarning: false,
+          editId: editId!,
+          packageName: packageName,
+          media: {
+            body: createReadStream(filepath),
+          },
+        },
+        {
+          timeout: 160000,
```

Comment:
Do you mind moving this to a constant somewhere?

(Also, was there any logic behind picking 160s? 2 and 2/3 minutes seems like a bit of a random timeout.)

Reply:
Yes, my math was wrong. The documentation says to set it to a 2 to 3 minute timeout and I was aiming for three minutes. Updated to a const and the timeout.